third_party/2and3/cryptography/hazmat/primitives/asymmetric/dh.pyi

```diff
@@ -0,0 +1,68 @@
+from abc import ABCMeta, abstractmethod
+from typing import Optional
+
+from cryptography.hazmat.backends.interfaces import DHBackend
+from cryptography.hazmat.primitives.serialization import (
+    Encoding,
+    KeySerializationEncryption,
+    ParameterFormat,
+    PrivateFormat,
+    PublicFormat,
+)
+
+class DHParameters(metaclass=ABCMeta):
+    @abstractmethod
+    def generate_private_key(self) -> DHPrivateKey: ...
+    @abstractmethod
+    def parameter_bytes(self, encoding: Encoding, format: ParameterFormat) -> bytes: ...
+    @abstractmethod
+    def parameter_numbers(self) -> DHParameterNumbers: ...
+
+DHParametersWithSerialization = DHParameters
+
+class DHParameterNumbers:
+    p: int
+    g: int
+    q: int
```

Comment:
These three are using `cryptography.utils.read_only_property`, which is only a fancy way of using `@property`, so we could just do that here:

```python
@property
def p(self) -> int: ...
@property
def g(self) -> int: ...
@property
def q(self) -> int: ...
```